examples/platforms/nrf528xx/nrf52840/nrf52840.cmake

```diff
@@ -0,0 +1,279 @@
+elseif(${OT_BOOTLOADER} STREQUAL "UART")
+    set(LD_FILE "${CMAKE_CURRENT_SOURCE_DIR}/nrf52840/nrf52840_bootloader_uart.ld")
+elseif(${OT_BOOTLOADER} STREQUAL "BLE")
+    set(LD_FILE "${CMAKE_CURRENT_SOURCE_DIR}/nrf52840/nrf52840_bootloader_ble.ld")
+else()
+    set(LD_FILE "${CMAKE_CURRENT_SOURCE_DIR}/nrf52840/nrf52840.ld")
+endif()
+
+list(APPEND OT_PLATFORM_DEFINES
+    "OPENTHREAD_CORE_CONFIG_PLATFORM_CHECK_FILE=\"openthread-core-nrf52840-config-check.h\""
+)
+if(OT_BUILTIN_MBEDTLS)
+    list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_CONFIG_FILE=\"mbedtls-config.h\"")
+    if(${OT_MBEDTLS_THREADING})
+        list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_THREADING_C")
+        list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_THREADING_ALT")
+        list(APPEND OT_PUBLIC_INCLUDES
+            "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/include/software-only-threading"
+        )
+    endif()
+else()
+    list(APPEND OT_PLATFORM_DEFINES "MBEDTLS_CONFIG_FILE=\"nrf-config.h\"")
+    list(APPEND OT_PUBLIC_INCLUDES
+        "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/include"
+        "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/config"
+        "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/nrf_cc310_plat/include"
+    )
+endif()
+
+set(OT_PLATFORM_DEFINES ${OT_PLATFORM_DEFINES} PARENT_SCOPE)
+list(APPEND OT_PRIVATE_DEFINES "MBEDTLS_USER_CONFIG_FILE=\"nrf52840-mbedtls-config.h\"")
+list(APPEND OT_PUBLIC_INCLUDES
+    "${PROJECT_SOURCE_DIR}/third_party/NordicSemiconductor/libraries/nrf_security/mbedtls_plat_config"
+)
+set(OT_PUBLIC_INCLUDES ${OT_PUBLIC_INCLUDES} PARENT_SCOPE)
+string(REPLACE "-pedantic-errors" "" OT_CFLAGS "${OT_CFLAGS}")
+string(REPLACE "-Wundef" "" OT_CFLAGS "${OT_CFLAGS}")
+list(APPEND OT_PLATFORM_DEFINES "OPENTHREAD_PROJECT_CORE_CONFIG_FILE=\"${OT_CONFIG}\"")
+
+add_library(openthread-nrf52840
+    ${NRF_COMM_SOURCES}
+    ${NRF_SINGLEPHY_SOURCES}
+    $<TARGET_OBJECTS:openthread-platform-utils>
+)
+
+add_library(openthread-nrf52840-transport
+    ${NRF_TRANSPORT_SOURCES}
+    $<TARGET_OBJECTS:openthread-platform-utils>
+)
+
+add_library(openthread-nrf52840-sdk
+    ${NRF_COMM_SOURCES}
+    ${NRF_SINGLEPHY_SOURCES}
+    $<TARGET_OBJECTS:openthread-platform-utils>
+)
+
+add_library(openthread-nrf52840-softdevice-sdk
+    ${NRF_COMM_SOURCES}
+    ${NRF_SOFTDEVICE_SOURCES}
+    $<TARGET_OBJECTS:openthread-platform-utils>
+)
+
+set_property(TARGET openthread-nrf52840 PROPERTY C_STANDARD 99)
+set_property(TARGET openthread-nrf52840-transport PROPERTY C_STANDARD 99)
+set_property(TARGET openthread-nrf52840-sdk PROPERTY C_STANDARD 99)
+set_property(TARGET openthread-nrf52840-softdevice-sdk PROPERTY C_STANDARD 99)
+if(NOT OT_BUILTIN_MBEDTLS)
+    target_link_libraries(openthread-nrf52840
+        PRIVATE
+            openthread-platform-utils
+        PUBLIC
+            nordicsemi-mbedtls
```

Comment:
set variable `OT_MBEDTLS` to this target, and use `OT_MBEDTLS` exclusively everywhere.